File: decorators.py
from pythonscript import js
js('var py = require("./py")')
print "Testing decorators"

def inc(fn):
    def the_fn(x):
        return fn(x) + 1
    return the_fn

@inc
@inc
def foo(x):
    return x

print "foo(1) is", foo(1)

File: js/decorators.js
var py = require("./py");
console.log ("Testing decorators");
inc = py.def ({}, function (fn) {
    the_fn = py.def ({}, function (x) {
        return fn ([x], {}) + 1;
    });
    return the_fn;
});
foo = inc ([inc ([py.def ({}, function (x) {
    return x;
})], {})], {});
console.log ("foo(1) is", foo ([1], {}));

